pkg/virt-controller/services/renderresources.go

```diff
@@ -199,7 +199,7 @@ func WithCPUPinning(cpu *v1.CPU) ResourceRendererOption {
 			emulatorThreadCPU := resource.NewQuantity(1, resource.BinarySI)
 			limits := renderer.calculatedLimits[k8sv1.ResourceCPU]
 			limits.Add(*emulatorThreadCPU)
-			renderer.calculatedLimits[k8sv1.ResourceCPU] = limits
+			renderer.vmLimits[k8sv1.ResourceCPU] = limits
```

Comment:
I must admit I don't understand the difference but it seems it's not the right thing.

Reply:
It is the right thing. I just messed up the test, I think.

Reply:
I am slightly confused as well :) This _is_ the CPU pinning case. 

Also, it should matter how CPUs are being requested; either by specifying the guest topology (preferably) or by explicitly setting `resources.requests.cpus`

Perhaps I misunderstood your comment. Could you please explain?

Reply:
Apologies I had stupidly confused the CPU and Memory validation checks for `dedicatedCPUPlacement` from my time working on https://github.com/kubevirt/kubevirt/issues/8867.